src/Node.cpp

```diff
@@ -53,15 +54,16 @@ Node::AddToParents(GroupWeak& aParent, Node& aChild) {
 }
 
 void
-Node::RemoveFromParents(GroupWeak& aParent, Node& aChild) {
-  GroupPtr parent = aParent.lock();
-  if (!parent) {
-    return;
-  }
-  for (auto it = aChild.m.parents.begin(); it != aChild.m.parents.end(); it++) {
-    if (it->lock() == parent) {
-      aChild.m.parents.erase(it);
+Node::RemoveFromParents(Group& aParent, Node& aChild) {
+  for (auto it = aChild.m.parents.begin(); it != aChild.m.parents.end();) {
+    Group* node = it->lock().get();
```

Comment:
It would be safer to create a extra variable with the lock result, and the do the get().

Reply:
I thought as long as it != end(), *it will always return a valid object? Is that not the case?

Reply:
My concern was about accessing `Group* node` contents without a shared_str protecting the value (the one returned from lock() get´s destroyed). But if we are only using the value for comparations and not actually using it, it's ok.